Hardware/Hardware/CardReader/StripeCardReader/UnderlyingError+Stripe.swift

```diff
@@ -0,0 +1,84 @@
+import StripeTerminal
+/// Maps Stripe SDK specific errors to domain errors:
+/// the mapping is done according to the error codes documented here:
+/// https://stripe.dev/stripe-terminal-ios/docs/Enums/SCPError.html
+extension UnderlyingError {
+    static func make(with stripeError: NSError) -> Self {
```

Comment:
I think having this as an initializer, could read a bit swifter, what do you think?
Also you could receive the stripe error and cast it here, to avoid doing the cast to `NSError` in every call site.

Reply:
Good points!